Modules/Core/Common/include/itkImageSink.hxx

```diff
@@ -0,0 +1,287 @@
+           !inputPtr1->GetDirection().GetVnlMatrix().as_ref().is_equal(inputPtrN->GetDirection().GetVnlMatrix(), this->m_DirectionTolerance) )
+        {
+        std::ostringstream originString, spacingString, directionString;
+        if ( !inputPtr1->GetOrigin().GetVnlVector().is_equal(inputPtrN->GetOrigin().GetVnlVector(), coordinateTol) )
+          {
+          originString.setf( std::ios::scientific );
+          originString.precision( 7 );
+          originString << "InputImage Origin: " << inputPtr1->GetOrigin()
+                       << ", InputImage" << it.GetName() << " Origin: " << inputPtrN->GetOrigin() << std::endl;
+          originString << "\tTolerance: " << coordinateTol << std::endl;
+          }
+        if ( !inputPtr1->GetSpacing().GetVnlVector().is_equal(inputPtrN->GetSpacing().GetVnlVector(), coordinateTol) )
+          {
+          spacingString.setf( std::ios::scientific );
+          spacingString.precision( 7 );
+          spacingString << "InputImage Spacing: " << inputPtr1->GetSpacing()
+                        << ", InputImage" << it.GetName() << " Spacing: " << inputPtrN->GetSpacing() << std::endl;
+          spacingString << "\tTolerance: " << coordinateTol << std::endl;
+          }
+        if ( !inputPtr1->GetDirection().GetVnlMatrix().as_ref().is_equal(inputPtrN->GetDirection().GetVnlMatrix(), this->m_DirectionTolerance) )
+          {
+          directionString.setf( std::ios::scientific );
+          directionString.precision( 7 );
+          directionString << "InputImage Direction: " << inputPtr1->GetDirection()
+                          << ", InputImage" << it.GetName() << " Direction: " << inputPtrN->GetDirection() << std::endl;
+          directionString << "\tTolerance: " << this->m_DirectionTolerance << std::endl;
+          }
+        itkExceptionMacro(<< "Inputs do not occupy the same physical space! "
+                          << std::endl
+                          << originString.str() << spacingString.str()
+                          << directionString.str() );
+        }
+      }
+    }
+
+}
+
+template <class TInputImage>
+void
+ImageSink<TInputImage>
+::StreamedGenerateData( unsigned int inputRequestedRegionNumber )
+{
+
+  this->GetMultiThreader()->SetNumberOfWorkUnits( this->GetNumberOfWorkUnits() );
+
+  // calculate the progress range for this streamed chunk
+  const ThreadIdType  total = this->GetNumberOfInputRequestedRegions();
+  const float oldProgress = float(inputRequestedRegionNumber)/(total);
+  const float newProgress = float(inputRequestedRegionNumber+1)/(total);
+  ProgressTransformer pt( oldProgress, newProgress, this );
+
+
+  this->GetMultiThreader()->template ParallelizeImageRegion<InputImageDimension>(
+    this->m_CurrentInputRegion,
+    [this](const InputImageRegionType & inputRegionForThread)
+    { this->ThreadedStreamedGenerateData(inputRegionForThread); },
+    pt.GetProcessObject());
```

Comment:
@dzenanz This was updated further to use the `ParallelizeImageRegion` method. The internal splitter is just used for the streamed region splitting now.